.github/workflows/nightly.yml

```diff
@@ -3,11 +3,14 @@ on:
   schedule:
    - cron: "0 1 * * *"
   workflow_dispatch: ~
+permissions: {} # none
```

Comment:
For this one I would find it reasonable to just define the read permissions in the general section, instead of each job. This is open source, after all. Not much that can be stolen.

Reply:
Done.

Reply:
for consistency, it should be done on all places

Reply:
In other places all that is needed is `issues` and/or `pull-requests` write permission. I think giving `contents: read` is unnecessary. But if you still think it is better for consistency, please let me know.

Reply:
yes, I belive `read` is better than a comment and than 3 different states

Reply:
Done. Take a look.